packages/logger/src/logger.test.ts

```diff
@@ -0,0 +1,96 @@
+import { describe, expect, it, vi } from "vitest";
+import { createLogger } from "./logger";
+import { attachLoggerConsoleTransport } from "./logger-console-transport";
+import { attachLoggerOtelTransport } from "./logger-otel-transport";
+import { logs } from "@opentelemetry/api-logs";
+import { SemanticResourceAttributes } from "@opentelemetry/semantic-conventions";
+
+vi.spyOn(console, "log");
+vi.setSystemTime(new Date(2024, 1, 1, 5, 15));
+
+describe("Logger", () => {
+  describe("Console Transport", () => {
+    it("Prints message and nested object to the console, including attributes passed from the parent scope", () => {
+      const logger = createLogger("Test Logger", {
+        rootScopePrimitiveArg: 1,
+        rootScopeObjectArg: {
+          objectKey: "objectValue",
+        },
+      });
+
+      attachLoggerConsoleTransport(logger);
+
+      logger.info("Test Message", {
+        childScopePrimitiveArg: 2,
+        childScopeObjectArg: {
+          objectKey: "objectValue",
+        },
+      });
+
+      /**
+       * TODO this assertion fails, but strings are the same. Why?
```

Comment:
**Suggestion:** this TODO can be removed

Reply:
done